convert2rhel/grub.py

```diff
@@ -572,6 +581,51 @@ def post_ponr_set_efi_configuration():
         _log_critical_error(e.message)
 
 
+def update_grub_after_conversion():
+    """Update GRUB2 images and config after conversion.
+
+    This is mainly a protective measure to prevent issues in case the original distribution GRUB2 tooling
+    generates images that expect different format of a config file. To be on the safe side we
+    rather re-generate the GRUB2 config file and install the GRUB2 image.
+    We opted for doing that only for GRUB2 and not for GRUB Legacy intentionally for the reason of
+    RHEL 6 having transitioned to the Extended Life-cycle Support phase.
+    """
+
+    if systeminfo.system_info.version.major == 6:
+        logger.warning(
+            "Convert2RHEL does not install updated GRUB Legacy bootloader image on RHEL 6. Install the image manually "
+            "by following https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/installation_guide/sect-grub-installing. "
+        )
+        return
+
+    utils.RestorableFile(GRUB2_BIOS_CONFIG_FILE).backup()
+    utils.RestorableFile(GRUB2_BIOS_ENV_FILE).backup()
+
+    grub2_config_file = GRUB2_BIOS_CONFIG_FILE if not is_efi() else os.path.join(RHEL_EFIDIR_CANONICAL_PATH, "grub.cfg")
+
+    _, ret_code = utils.run_subprocess(["/usr/sbin/grub2-mkconfig", "-o", grub2_config_file], print_output=False)
+
+    if ret_code != 0:
+        logger.warning("GRUB2 config file generation failed.")
```

Comment:
Might be beneficial to add another debug log for the output for both `grub2-mkconfig` commands